src/main/java/org/jboss/windup/operator/KubernetesClientProducer.java

```diff
@@ -0,0 +1,68 @@
+package org.jboss.windup.operator;
+
+import io.fabric8.kubernetes.api.model.apiextensions.v1beta1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apiextensions.v1beta1.DoneableCustomResourceDefinition;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.kubernetes.internal.KubernetesDeserializer;
+import io.quarkus.arc.profile.IfBuildProfile;
+import lombok.extern.java.Log;
+import org.jboss.windup.operator.model.WindupResource;
+import org.jboss.windup.operator.model.WindupResourceDoneable;
+import org.jboss.windup.operator.model.WindupResourceList;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.io.InputStream;
+
+@Log
+public class KubernetesClientProducer {
+    @Produces
+    @Singleton
+    @Named("namespace")
+    String findMyCurrentNamespace(NamespacedKubernetesClient client)  {
+        return client.getConfiguration().getNamespace();
+    }
+
+    @Produces
+    @Singleton
+    @IfBuildProfile("prod")
+    NamespacedKubernetesClient makeDefaultClient() {
+        log.info("Creating K8s Client instance");
+        return new DefaultKubernetesClient();
+    }
+
+    @Produces
+    @Singleton
+    MixedOperation<WindupResource, WindupResourceList, WindupResourceDoneable, Resource<WindupResource, WindupResourceDoneable>>
+    makeWindupCustomResource(NamespacedKubernetesClient defaultClient, CustomResourceDefinitionContext crdContext, @Named("namespace") String namespace) {
+        log.info("Registering custom kind");
+        KubernetesDeserializer.registerCustomKind("windup.jboss.org/v1", "Windup", WindupResource.class);
+
+        return defaultClient.inNamespace(namespace).customResources(crdContext, WindupResource.class, WindupResourceList.class, WindupResourceDoneable.class);
+    }
+
+    @Produces
+    @Singleton
+    CustomResourceDefinitionContext makeCRDContext(NamespacedKubernetesClient defaultClient, @Named("namespace") String namespace) {
+        log.info("Creating windup.crd.yaml stream");
+        InputStream fileStream = KubernetesClientProducer.class.getResourceAsStream("/k8s/def/windup.crd.yaml");
+
+        log.info("Loading windup.crd.yaml");
+        log.info("client : " + defaultClient);
+        log.info("crds : " + defaultClient.customResourceDefinitions());
+        Resource<CustomResourceDefinition, DoneableCustomResourceDefinition> resource = defaultClient.inNamespace(namespace).customResourceDefinitions().load(fileStream);
+        log.info("resource : " + resource);
+        CustomResourceDefinition windupCRD = resource.get();
```

Comment:
`CustomResourceDefinition` is imported from package `v1beta1` and then loaded from `windup.crd.yaml` which has [`apiVersion: apiextensions.k8s.io/v1`](https://github.com/windup/windup-operator/pull/1/files#diff-a16ac53c4f98971fade0aeb3c518b1ad815b5c0020bdd765c6eafb112f32a3a4R1).
Shouldn't we keep the versions to be the same?

Reply:
:+1: